src/thread/thread_worker_pool.hpp

```diff
@@ -1460,4 +1460,90 @@ namespace cubthread
 
 } // namespace cubthread
 
+
+namespace cubthread
+{
+  template <typename Context>
+  class worker_pool_task_capper
+  {
+      using context_type = Context;
+      using task_type = task<Context>;
+      using worker_pool_type = worker_pool<Context>;
+
+    public:
+      worker_pool_task_capper (context_manager<Context> *manager, unsigned int pool_size, char *worker_pool_name);
+      ~worker_pool_task_capper ();
+
+      void push_task (task<Context> *task);
+      cubthread::worker_pool<Context> *get_worker_pool ();
+      void end_task ();
+
+    private:
+      cubthread::worker_pool<Context> *m_worker_pool;
+      unsigned int m_tasks_available;
+      unsigned int m_pool_size;
+      std::mutex m_mutex;
+      std::condition_variable m_cond_var;
+  };
+} // namespace cubthread
+
+namespace cubthread
+{
+  //////////////////////////////////////////////////////////////////////////
+  // Blocking manager
+  //////////////////////////////////////////////////////////////////////////
+  template <typename Context>
+  worker_pool_task_capper<Context>::worker_pool_task_capper (context_manager<Context> *manager, unsigned int pool_size,
```

Comment:
Don't create/destroy worker pool inside worker_pool_task_capper. Better to construct worker_pool_task_capper based on pre-existing worker pool.